api/integreatly/v1alpha1/grafanafolder_types.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GrafanaPermissionItem struct {
+	PermissionTargetType string `json:"permissionTargetType"`
+	PermissionTarget     string `json:"permissionTarget"`
+	PermissionLevel      int    `json:"permissionLevel"`
+}
+
+type GrafanaFolderSpec struct {
+	FolderName        string                  `json:"title"`
+	FolderPermissions []GrafanaPermissionItem `json:"permissions,omitempty"`
```

Comment:
It is kind of self explanatory but it would be nice with some extra comments explaining what the different CRD values does.
Maybe copy the description from the grafana API docs.